libs/store/src/lib/domain/decorators.ts

```diff
@@ -1,52 +1,54 @@
 import 'reflect-metadata';
 
-import { EntityMetaOptions } from './entity';
+import { BaseEntity } from './entity';
+import { EntityMetaOptions } from './entity-meta-options';
 import { EntityResourceMetaOptions } from './rest-resource';
+import { Type } from './utility';
 
 export const ENTITY_KEY = Symbol('ENTITY');
 export const REST_RESOURCE_KEY = Symbol('REST_Resource');
 
 const entitiesSet = new Set();
 
-export type Type<T> = new (...args: any[]) => T;
+export type EntityType<T extends BaseEntity> = Type<T>;
 
-export function Entity(config: EntityMetaOptions) {
+export function Entity<T extends BaseEntity>(config: EntityMetaOptions<T>) {
     // TODO: Add entity name uniqueness validation
-    return <T>(target: Type<T>) => {
+    return (target: EntityType<T>) => {
         if (!entitiesSet.has(target)) {
             entitiesSet.add(target);
         }
         Reflect.defineMetadata(ENTITY_KEY, config, target);
     };
 }
 
-export function RESTResource(config: EntityResourceMetaOptions) {
-    return <T>(target: Type<T>) => {
+export function RESTResource<T extends BaseEntity>(config: EntityResourceMetaOptions) {
+    return (target: EntityType<T>) => {
         if (!entitiesSet.has(target)) {
             entitiesSet.add(target);
         }
         Reflect.defineMetadata(REST_RESOURCE_KEY, config, target);
```

Comment:
Is this Reflect always available? Should we rely on it? Or maybe do something like angular in his code? 

<img width="714" alt="Screen Shot 2021-04-07 at 9 11 30 AM" src="https://user-images.githubusercontent.com/17149942/113825490-52c2b700-9781-11eb-8bec-ca432cbcd763.png">

Reply:
Removed `reflect-metadata` and switched to the plain `Map` usage